analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParametersCorrectOrder.cs

```diff
@@ -193,4 +193,41 @@ public void M3()
             double doTheThing(int divisor, int dividend) => divide(dividend, divisor);  // Noncompliant
         }
     }
+
+    // See https://github.com/SonarSource/sonar-dotnet/issues/3879
```

Comment:
This is C# and VB.NET rule. Can you please try to implement the same logic for VB.NET as well? In VB.NET, the syntaxes are:
```
a? As Integer
a As Integer?
a As Nullable(Of Integer)

//and then as usually
a.HasValue 
a.Value
```

Reply:
Added VB.NET implementation

Reply:
Please use also those `a?` and `Integer?` examples in the test cases to cover all variants (same as for C# above)